### DISCUSSION TITLE
How to Obtain the Width and  Height of the SplitColumn Layout When Using Spectre.Console?

### DISCUSSION THREAD

**Discussion by yingpanwang** [spectreconsole/spectre.console#1693] [Q&A] [answered]
I'm currently creating a demo for a console game using the Spectre.Console library. During implementation, I encountered a problem: I cannot determine the width and height of the "main" area of the console. Could anyone advise on how I should write the code to dynamically get the dimensions of the console's main area?

i‘m using a Canvas to Display the Tested Drawing Area in Spectre.Console
e.g:

```chsarp


// 初始化布局
using Spectre.Console;

var main = new Layout("Main") { Size = 80 };

var sider = new Layout("Right") { Size = 20 }
           .SplitRows(
               new Layout("Top"),
               new Layout("Bottom"));

var layout = new Layout("Root")
    .SplitColumns(
        main, sider
       );

// 创建实时更新
AnsiConsole.Live(Align.Center(layout))
    .Start(ctx =>
    {
        // 使用 Canvas 填充 main 区域
        int terminalWidth = Console.WindowWidth;
        int terminalHeight = Console.WindowHeight;

        int width = terminalWidth / 2, height = terminalHeight;
        var canvas = new Canvas(width, height);
        //main.Update(Align.Center(new Markup($" width:{width} ,height:{height}"), VerticalAlignment.Middle));

        for (int x = 0; x < width; x++)
        {
            for (int y = 0; y < height; y++)
            {
                canvas.SetPixel(x, y, Color.White);

                main.Update(
                        Align.Center(canvas, VerticalAlignment.Middle)
                    );

                // 更新布局
                ctx.Refresh();
            }
        }
    });

await Console.In.ReadLineAsync();


```

![image](https://github.com/user-attachments/assets/e73845a1-5e9d-4bca-ba6b-6bd492720984)

**Comment by patriksvensson** [comment, marked as answer]
You can't get the width and the height like that, since it's calculated dynamically on rendering. I think your best bet is to create a custom `IRenderable`, that way you can get the requested max width. The height you cannot know, but if you're running in "full screen" you can probably use that as height.
